File: ALittle/Module/ALittleIDE/Script/UIEdit/IDEEditCore/IDESetting/ScrollBarS.lua
-- ALittle Generate Lua
module("ALittleIDE", package.seeall)

local ___thispath = select('1', ...):match("(.+[/\\]).+$") or ""
local ___rawset = rawset
local ___pairs = pairs
local ___ipairs = ipairs


assert(ALittleIDE.Grid3S, " extends class:ALittleIDE.Grid3S is nil")
ScrollBarS = Lua.Class(ALittleIDE.Grid3S, "ALittleIDE.ScrollBarS")

function ScrollBarS:Ctor(user_info, tab_child, tree_logic)
	___rawset(self, "_layer_name", "ide_setting_scrollbar")
end

function ScrollBarS:LoadNatureBase()
	Grid3S.LoadNatureBase(self)
	self:LoadValueData("total_size")
	self:LoadValueData("show_size")
	self:LoadValueData("offset_rate")
	self:LoadValueData("offset_step")
	self:LoadShowTypeDataForExtends("up_button")
	self:LoadShowTypeDataForExtends("down_button")
	self:LoadShowTypeDataForExtends("bar_button")
	self:LoadShowTypeDataForImage("bar_background")
end

function ScrollBarS:HandleTotleSizeFOCUSOUT(event)
	self:ValueNumInputChange("total_size", false)
end

function ScrollBarS:HandleShowSizeFOCUSOUT(event)
	self:ValueNumInputChange("show_size", false)
end

function ScrollBarS:HandleOffsetRateFOCUSOUT(event)
	self:Z2OValueInputChange("offset_rate", false)
end

function ScrollBarS:HandleOffsetStepFOCUSOUT(event)
	self:Z2OValueInputChange("offset_step", false)
end

function ScrollBarS:HandleUpButtonFOCUSOUT(event)
	if event.target._user_data ~= nil then
		if event.target._user_data == event.target.text then
			return
		end
		event.target._user_data = event.target.text
	end
	self:RemoverToNilShowSetForExtends("up_button", self._up_button.text, false)
end

function ScrollBarS:HandleDownButtonFOCUSOUT(event)
	if event.target._user_data ~= nil then
		if event.target._user_data == event.target.text then
			return
		end
		event.target._user_data = event.target.text
	end
	self:RemoverToNilShowSetForExtends("down_button", self._down_button.text, false)
end

function ScrollBarS:HandleBarButtonFOCUSOUT(event)
	if event.target._user_data ~= nil then
		if event.target._user_data == event.target.text then
			return
		end
		event.target._user_data = event.target.text
	end
	self:RemoverToNilShowSetForExtends("bar_button", self._bar_button.text, false)
end

function ScrollBarS:HandleBarBackFOCUSOUT(event)
	if event.target._user_data ~= nil then
		if event.target._user_data == event.target.text then
			return
		end
		event.target._user_data = event.target.text
	end
	self:RemoverToNilShowSetForImage("bar_background", self._bar_background.text, self._bar_background_grid9.selected, false)
end

function ScrollBarS:HandleBarBackSelect(event)
	local func = Lua.Bind(self.ImagePathSelectCallback, self, "bar_background", self.HandleBarBackFOCUSOUT, nil)
	g_IDEAttrImageDialog:ShowDialog(nil, func)
end

File: ALittleIDE/Module/ALittleIDE/JSScript/IDE/UIEdit/Setting/ScrollBarS.js
{
if (typeof ALittleIDE === "undefined") window.ALittleIDE = {};


if (ALittleIDE.Grid3S === undefined) throw new Error(" extends class:ALittleIDE.Grid3S is undefined");
ALittleIDE.ScrollBarS = JavaScript.Class(ALittleIDE.Grid3S, {
	Ctor : function(user_info, tab_child, tree_logic) {
		this._layer_name = "ide_setting_scrollbar";
	},
	LoadNatureBase : function() {
		ALittleIDE.Grid3S.LoadNatureBase.call(this);
		this.LoadValueData("total_size");
		this.LoadValueData("show_size");
		this.LoadValueData("offset_rate");
		this.LoadValueData("offset_step");
		this.LoadShowTypeDataForExtends("up_button");
		this.LoadShowTypeDataForExtends("down_button");
		this.LoadShowTypeDataForExtends("bar_button");
		this.LoadShowTypeDataForImage("bar_background");
	},
	HandleTotleSizeFOCUSOUT : function(event) {
		this.ValueNumInputChange("total_size", false);
	},
	HandleShowSizeFOCUSOUT : function(event) {
		this.ValueNumInputChange("show_size", false);
	},
	HandleOffsetRateFOCUSOUT : function(event) {
		this.Z2OValueInputChange("offset_rate", false);
	},
	HandleOffsetStepFOCUSOUT : function(event) {
		this.Z2OValueInputChange("offset_step", false);
	},
	HandleUpButtonFOCUSOUT : function(event) {
		if (event.target._user_data !== undefined) {
			if (event.target._user_data === event.target.text) {
				return;
			}
			event.target._user_data = event.target.text;
		}
		this.RemoverToNilShowSetForExtends("up_button", this._up_button.text, false);
	},
	HandleDownButtonFOCUSOUT : function(event) {
		if (event.target._user_data !== undefined) {
			if (event.target._user_data === event.target.text) {
				return;
			}
			event.target._user_data = event.target.text;
		}
		this.RemoverToNilShowSetForExtends("down_button", this._down_button.text, false);
	},
	HandleBarButtonFOCUSOUT : function(event) {
		if (event.target._user_data !== undefined) {
			if (event.target._user_data === event.target.text) {
				return;
			}
			event.target._user_data = event.target.text;
		}
		this.RemoverToNilShowSetForExtends("bar_button", this._bar_button.text, false);
	},
	HandleBarBackFOCUSOUT : function(event) {
		if (event.target._user_data !== undefined) {
			if (event.target._user_data === event.target.text) {
				return;
			}
			event.target._user_data = event.target.text;
		}
		this.RemoverToNilShowSetForImage("bar_background", this._bar_background.text, this._bar_background_grid9.selected, false);
	},
	HandleBarBackSelect : async function(event) {
		let ui_manager = ALittleIDE.g_IDEProject.GetUIManager(this._tree_logic.user_info.module);
		if (ui_manager === undefined) {
			g_AUITool.ShowNotice("错误", "模块不存在:" + this._tree_logic.user_info.module);
			return;
		}
		ALittleIDE.g_IDEImageSelectDialog.SetBasePath(ui_manager.texture_path);
		let path = await ALittleIDE.g_IDEImageSelectDialog.ShowSelect();
		if (path === undefined) {
			return;
		}
		this.ImagePathSelectCallback("bar_background", this.HandleBarBackFOCUSOUT, undefined, path, false);
	},
}, "ALittleIDE.ScrollBarS");

}